python/cuml/_thirdparty/sklearn/preprocessing/_imputation.py

```diff
@@ -379,27 +379,40 @@ def _dense_fit(self, X, strategy, missing_values, fill_value):
             count_missing_values = mask.sum(axis=0)
             n_elems = X.shape[0] - count_missing_values
             mean = np.nansum(X, axis=0)
-            mean -= (count_missing_values * missing_values)
+            if not np.isnan(missing_values):
+                mean -= (count_missing_values * missing_values)
```

Comment:
This makes it so that we are implicitly counting the missing values as `0` for calculating the mean, is that the expected behavior? (can't recall off the top of my head). If not, then `n_elems` needs to be taken into account here

Reply:
Yep, @viclafargue explained it better than I could have!